sdk/trace/src/main/java/io/opentelemetry/sdk/trace/internal/ExtendedSpanProcessor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.trace.internal;
+
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+
+/**
+ * A specialization of {@link SpanProcessor} providing more extension points.
+ *
+ * <p>Note that this interface is considered experimental and therefore should only be used at the
+ * risk of its methods being changed or removed. If it stabilized, the interface is likely removed
+ * and the methods are promoted to {@link SpanProcessor}.
```

Comment:
We have standard boiler plate for this:
```
 * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
 * at any time.
```
But I wonder if we should adjust our boilerplate for these types of methods - internal because we make no compatibility guarantees, but acceptable for public consumption provided the user accepts the risks of breaking changes.

Maybe something like:
```
This class is internal and experimental. Its APIs are unstable and can change at any time. Its APIs (or a version of them) may be promoted to the public stable API in the future, but no guarantees are made.
```

Places this updated message would be relevant:
- SdkTracerProviderUtil
- SdkMeterProviderUtil
- MeterConfig, TracerConfig, LoggerConfig
- SdkEventLoggerProvider
- ... and more

Reply:
Yeah I found the `hence not for public use` a bit to strict for users, that's why a though a change might be make sense here. However, I didn't know that there were already other similar cases in the `internal` package.

I guess for this PR it is easiest to just stick to the existing standard paragraph and change it later everywhere if desired?

Reply:
> I guess for this PR it is easiest to just stick to the existing standard paragraph and change it later everywhere if desired?

Let's do that. Opened #6662 to track.